File: Qt_turtle_service/demo/ROS_demo/turtle_demo/src/turtle_circle.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc, argv, "turtle_circle");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 1000);

    geometry_msgs::Twist msg;
    msg.linear.x = 1.0;
    msg.linear.y = 0.0;
    msg.linear.z = 0.0;

    msg.angular.x = 0.0;
    msg.angular.y = 0.0;
    msg.angular.z = 2.0;

    //4-2.设置发送频率
    ros::Rate r(10);

    while (ros::ok())
    {
        /* code */
        pub.publish(msg);
        ros::spinOnce();
    }
    
    return 0;
}

File: ROS_demo/turtle_demo/scripts/turtle_circle.py
#! /usr/bin/env python
# coding: utf-8

import rospy
from geometry_msgs.msg import Twist

if __name__ == "__main__":

    node_name = "turtle_circle_py"

    rospy.init_node(node_name)

    topic_name = "/turtle1/cmd_vel"
    pub = rospy.Publisher(topic_name, Twist, queue_size=1000)

    rate = rospy.Rate(10)

    msg = Twist()
    msg.linear.x = 1.0
    msg.linear.y = 0.0
    msg.linear.z = 0.0
    msg.angular.x = 0.0
    msg.angular.y = 0.0
    msg.angular.z = 0.5

    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep()